src/QATools/Example/Pages/LoginPage.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * This file is part of the QA-Tools library.
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ *
+ * @copyright Michael Iwersen <mi.iwersen@gmail.com>
+ * @link      https://github.com/qa-tools/behat-example
+ */
+
+namespace QATools\Example\Pages;
+
+
+use QATools\QATools\HtmlElements\Element\Form;
+use QATools\QATools\HtmlElements\TypifiedPage;
+
+/**
+ * Class LoginPage.
+ *
+ * @page-url('/')
+ */
+class LoginPage extends TypifiedPage
+{
+
+	/**
+	 * The login form.
+	 *
+	 * @var Form
+	 * @find-by('id' => 'login-form')
+	 */
+	protected $loginForm;
+
+	/**
+	 * Fill the login form with given data.
+	 *
+	 * @param string $username The username.
+	 * @param string $password The password.
+	 *
+	 * @return static
+	 */
+	public function fillForm($username, $password)
+	{
+		$this->loginForm->fill(array(
+			'username' => $username,
+			'password' => $password,
+		));
+
+		return $this;
+	}
+
+	/**
+	 * Does the login.
+	 *
+	 * @return Page
+	 */
+	public function loginSuccess()
```

Comment:
By showing implementation details (login form has 2 fields and it can be submitted) as different feature context steps we're breaking encapsulation of PageObject class. In non-Gherkin based test suite, like PHPUnit separation of submit and fill methods isn't needed at all.

> Given the user logins with 'username' and 'password'
And the login is failing
Then the error message should say 'invalid username or password'

This could be:
```gherkin
Given the user logins with 'username' and 'password'
Then login should fail with 'invalid username or password' error message
```

I'm not sure if we can skip `When` part here.